analysis_options.yaml

```diff
@@ -4,10 +4,6 @@
 
 # Root analysis options shared among all Dart code in the respository. Based
 # on the Fuchsia standard analysis options, with some changes.
-analyzer:
```

Comment:
FWIW, there are still strong mode options that aren't the default (see implicit-cast and implicit-dynamic). We're considering trying to turn those on for Variant, but they trigger a lot of analysis failures right now.

Reply:
https://www.dartlang.org/guides/language/analysis-options

Reply:
Thanks for the pointer! I tried turning them on, and fixed some of the more obvious errors (added to this PR), but some of them were more fiddly, and there aren't easy-to-find example solutions for some of them, so I'm leaving them off for now.